crates/superposition_types/src/custom_query.rs

```diff
@@ -199,6 +200,49 @@ impl<'de> Deserialize<'de> for PaginationParams {
     }
 }
 
+#[derive(Debug, Deserialize, Clone, Deref)]
+#[deref(forward)]
+pub struct StringArgs(
+    #[serde(deserialize_with = "deserialize_stringified_list")] pub Vec<String>,
+);
+
+pub fn deserialize_stringified_list<'de, D, I>(
+    deserializer: D,
+) -> std::result::Result<Vec<I>, D::Error>
+where
+    D: de::Deserializer<'de>,
+    I: de::DeserializeOwned,
+{
+    struct StringVecVisitor<I>(std::marker::PhantomData<I>);
+
+    impl<'de, I> de::Visitor<'de> for StringVecVisitor<I>
+    where
+        I: de::DeserializeOwned,
+    {
+        type Value = Vec<I>;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            formatter.write_str(
+                "a string containing comma separated values eg: CREATED,INPROGRESS",
+            )
+        }
+
+        fn visit_str<E>(self, v: &str) -> std::result::Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            let mut query_vector = Vec::new();
+            for param in v.split(',') {
+                let p: I = I::deserialize(param.into_deserializer())?;
+                query_vector.push(p);
+            }
+            Ok(query_vector)
+        }
+    }
+
+    deserializer.deserialize_any(StringVecVisitor(std::marker::PhantomData::<I>))
+}
+
```

Comment:
this should work, right ?
```suggestion
#[derive(Debug, Clone, Deref)]
#[deref(forward)]
pub struct StringArgs(pub Vec<String>);

impl<'de> Deserialize<'de> for StringArgs {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: Deserializer<'de>,
    {
        let s = String::deserialize(deserializer)?;
        let items = s.split(',').map(|item| item.trim().to_string()).collect();

        Ok(Self(items))
    }
}```

Reply:
In the case you present, if the user passes a wrong value then there is no error, which I want.

Reply:
this does return an error, if the data passed is not a string, which is more than enough to determine if the value is right or not
as the data in it must contain is not mandatory, because there can always be a single element in the array which does not require any sorts of delimiter

Reply:
@ayushjain17 I tried this, but this function is used to deserialize a lot of other vectors and type cast them. I don't want to manually impl deserialize on all of them so I won't change this for now.